Lib/glyphsLib/classes.py

```diff
@@ -1747,6 +1760,91 @@ def _set_metric(self, metricname, value):
             self.metrics.append(GSMetricValue(position=0))
         self.metrics[metricIndex] = GSMetricValue(position=value)
 
+    @property
+    def alignmentZones(self):
+        # If there are values that are parsed from file format 2 or which are
+        # explicitly set return those
+        if self._alignmentZones is not None:
+            return self._alignmentZones
+
+        if len(self.metrics) == 0:
+            return []
+
+        validZoneMetrics = (
+            "ascender",
+            "cap height",
+            "x-height",
+            "baseline",
+            "descender",
+        )
+
+        zones = []
+        for index, fontMetric in enumerate(self.font.metrics):
+            if fontMetric.type not in validZoneMetrics:
+                continue
+            metric = self.metrics[index]
+            zone = GSAlignmentZone(pos=metric.position, size=metric.overshoot)
+            zones.append(zone)
+        return zones
+
+    @alignmentZones.setter
+    def alignmentZones(self, entries):
+        if type(entries) is not list:
+            raise ValueError(
```

Comment:
usually it's preferable to use isinstance instead of `type() is ...`, unless you really have a good reason to restrict the exact type. And raise TypeError instead of ValueError when the type is not as expected